### PR TITLE
First try at adding links to responses.

### PR DESCRIPTION
**Pull request by TimothyBJacobs**
This is a first try at adding info about modifying responses to include custom links.

### CODE REVIEW

File: extending-the-rest-api/modifying-responses.md [hunk 1]
```diff
@@ -128,3 +128,88 @@ register_meta( $object_type, 'my_meta_key', $args1 );
 This example shows how to allow reading and writing of a post meta field. This will allow that field to be updated via a POST request to `wp-json/wp/v2/posts/<post-id>` or created along with a post via a POST request to `wp-json/wp/v2/posts/`.
 
 Note that for meta fields registered on custom post types, the post type must have `custom-fields` support. Otherwise the meta fields will not appear in the REST API.
+
+
+## Adding Links to the API Response
+WordPress generates a list of links associated with the queried resource to make it easier to navigate to related resources. The `embeddable` attribute controls whether the linked resource appears in the `_embedded` section of the repsonse.
```
Comment (rmccue) [inline extending-the-rest-api/modifying-responses.md]: I'd split the `embeddable` bit to later on and describe links by themselves first. I'd also probably note something like "links are sent to the client as the `_links` property of the JSON object, but are stored separately internally from the response data"

File: extending-the-rest-api/modifying-responses.md [hunk 2]
```diff
@@ -128,3 +128,88 @@ register_meta( $object_type, 'my_meta_key', $args1 );
...
+$response->add_link( 'author', rest_url( "/wp/v2/users/{$post->post_author}" ), array(
+	'embeddable' => true
+) );
+
+$response->add_link( 'https://api.w.org/term', add_query_arg( 'post', $post->ID, rest_url( "/wp/v2/{$tax_base}" ) ), array(
+	'embeddable' => true
+) );
+```
+
+The link relation MUST be either a [registered link relation from the IANA](https://www.iana.org/assignments/link-relations/link-relations.xhtml) or a URL that is under your control.
```
Comment (rmccue) [inline extending-the-rest-api/modifying-responses.md]: URL -> URI

File: extending-the-rest-api/modifying-responses.md [hunk 3]
```diff
@@ -128,3 +128,88 @@ register_meta( $object_type, 'my_meta_key', $args1 );
...
+  }
+}
+```
+_A sample of links from a Make.WordPress.org post_
+
+Custom links can be added to the response by using the `WP_REST_Response::add_link()` method. This method accepts three parameters, the link relation, the URL and optionally a list of link attributes. For example, to add the `author` and `wp:term` link.
+
+```php
+<?php
+$response->add_link( 'author', rest_url( "/wp/v2/users/{$post->post_author}" ), array(
```
Comment (rmccue) [inline extending-the-rest-api/modifying-responses.md]: I'd have these without `embeddable` for the first examples

**Comment by TimothyBJacobs** [comment]
@rmccue Updated.

**Comment by TimothyBJacobs** [comment]
@rmccue Bumped

File: extending-the-rest-api/modifying-responses.md [hunk 4]
```diff
@@ -128,3 +128,129 @@ register_meta( $object_type, 'my_meta_key', $args1 );
...
+
+### Registering a CURIE
+
+WordPress version 4.5 introduced support for Compact URIs, or CURIEs. This makes it possible to reference links by a much simpler identifier than the full URL which could easily be quite lengthy.
+
+A CURIE is registered using the `rest_response_link_curies` filter. For example.
+
+```php
+<?php
+function my_plugin_prefix_register_curie( $curies ) {
```
Comment (kadamwhite) [inline extending-the-rest-api/modifying-responses.md]: Needs a line in the example showing the actual filter registration.

**Pull request merged**